processor/code_processor.go

```diff
@@ -0,0 +1,74 @@
+package processor
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/NickGowdy/deveui-cli/client"
+	"github.com/NickGowdy/deveui-cli/device"
+)
+
+type CodeProcessor struct {
+	CodeRegistrationLimit int
+	MaxConcurrentJobs     int
+	LoraWAN               client.LoraWAN
+	DeviceCh              chan device.Device
+	DoneCh                chan struct{}
+}
+
+// Worker attempts to register a valid DevEUI via external LoRaWAN API.
+// If successful, a RegisterDevice struct with its Identifier and Code will be sent to the work channel.
+//
+// # Example
+//
+//	Identifier: 1CEB0080F074F750, Code: 4F750
+//
+// When an unexpected error occurs, return ctx.Err instead.
+func (cp *CodeProcessor) Worker(ctx context.Context, workCh chan device.Device, doneCh chan struct{}) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-workCh:
+			registeredDevice, err := registerDevice(cp.LoraWAN, ctx)
+			if err == nil {
+				cp.DeviceCh <- *registeredDevice
+			} else {
+				return err
+			}
+		case <-doneCh:
+			fmt.Printf("Done \n")
+			return nil
+		}
+	}
+}
+
+func registerDevice(loraWAN client.LoraWAN, ctx context.Context) (*device.Device, error) {
```

Comment:
It's more usual in Go for `context.Context` to be the first argument - most Go engineers would find this arrangement very unnatural.